jest.config.mjs

```diff
@@ -0,0 +1,36 @@
+// /*
+//  * For a detailed explanation regarding each configuration property, visit:
+//  * https://jestjs.io/docs/configuration
+//  */
+
+/** @type {import('ts-jest/dist/types').InitialOptionsTsJest} */
+export default {
+    projects: [
+        {
+            displayName: 'ESLint',
+            runner: 'jest-runner-eslint',
+            watchPlugins: ['jest-runner-eslint/watch-fix'],
+        },
+        {
+            displayName: 'ts-check and test',
+            extensionsToTreatAsEsm: ['.ts'],
+            preset: 'ts-jest/presets/default-esm',
+            moduleNameMapper: {
+                '^(\\.{1,2}/.*)\\.js$': '$1',
```

Comment:
I'm not sure I understand why we're adding `*.js` to file paths, especially since most files are Typescript. Also, since this is a mapping, does that mean that the paths without `*.js` are equivalent? I'm worried that the IDE's auto-import feature would add the path without `*.js`.

Reply:
For this: https://kulshekhar.github.io/ts-jest/docs/guides/esm-support

Long story short: With ESM, if you don't use a full path, it's invalid.
Jest's support of ESM OOTB ain't great, but fortunately, ts-jest does it for us.

Jest use 'CJS' resolution in its core (circus), so to help him out, we need to specify that 'yes, this local path ending in .js should be resolved by jest as this same path, stripped out of its `.js`.


I can understand that the decision to switch to ESM might seem to be overkill or too soon, but from my experience, it's worth it doing it now:

If we persist with CJS, we will end up like `coveo/cli`: a bunch of not updatable dependencies (importing ESM into CJS is a pain) and a big pain to migrate to ESM.

So instead of 'freezing' the dep, I decide to bite the bullet right away.


For the IDE auto import fret not
![image](https://user-images.githubusercontent.com/12366410/211610216-3e4f1372-b9d0-4d5f-819c-a2985cfb6093.png)